test/NuGet.Clients.Tests/NuGet.CommandLine.Test/RestoreTransitiveLoggingTests.cs

```diff
@@ -75,6 +75,65 @@ public void GivenAProjectReferenceNoWarnsVerifyNoWarning()
         }
 
         [Fact]
+        // Tests ProjA[AssemblyName=Test] -> ProjB[PkgX NoWarn NU1603] -> PkgX[NU1603]
+        public void GivenAProjectReferenceNoWarnsAndParentProjectContainsAssemblyNameVerifyNoWarning()
```

Comment:
There are no tests for the spec generation of legacy though. 

This will only test the .NET Core code path while the package spec is created in 3 different locations